magpie/ui/management/views.py

```diff
@@ -216,31 +218,59 @@ def add_user(self):
                 data = {"group_name": group_name}
                 resp = request_api(self.request, schemas.GroupsAPI.path, "POST", data=data)
                 if resp.status_code == HTTPConflict.code:
-                    return_data["conflict_group_name"] = True
+                    return_data["invalid_group_name"] = True
+                    return_data["reason_group_name"] = "Conflict"
             if user_email in self.get_user_emails():
-                return_data["conflict_user_email"] = True
+                return_data["invalid_user_email"] = True
+                return_data["reason_user_email"] = "Conflict"
             if user_email == "":
                 return_data["invalid_user_email"] = True
             if len(user_name) > get_constant("MAGPIE_USER_NAME_MAX_LENGTH"):
-                return_data["too_long_user_name"] = True
+                return_data["invalid_user_name"] = True
+                return_data["reason_user_name"] = "Too Long"
             if user_name in self.get_user_names():
-                return_data["conflict_user_name"] = True
+                return_data["invalid_user_name"] = True
+                return_data["reason_user_name"] = "Conflict"
             if user_name == "":
                 return_data["invalid_user_name"] = True
             if password is None or isinstance(password, six.string_types) and len(password) < 1:
                 return_data["invalid_password"] = True
-            elif password != confirm:
+            elif compare_digest(password, confirm):
```

Comment:
Looks like this function is causing the password comparaison issue (weird)

Reply:
Which python version are you running (version-dependent implementation)?

Reply:
oh, yes, hmm... help to add `not` in front of that compare-equal
patch coming up